utilities/transactions/write_prepared_txn.cc

```diff
@@ -91,6 +91,8 @@ Status WritePreparedTxn::CommitInternal() {
   auto s = db_impl_->WriteImpl(write_options_, working_batch, nullptr, nullptr,
                                log_number_, disable_memtable, &seq_used);
   uint64_t& commit_seq = seq_used;
+  // TODO(myabandeh): Reject a commit request if AddCommitted cannot encode
+  // commit_seq. This happens if prep_seq <<< commit_seq.
```

Comment:
I think we should put it into old_commit_map in that case?

Reply:
this is very rare case. we just need to put a safety guard here. as long as returning error is correct no need to complicated the impl.